code/modules/awaymissions/mission_code/ghost_role_spawners/golems.dm

```diff
@@ -134,6 +134,24 @@
 		user.death()
 		return
 
+/obj/effect/mob_spawn/human/golem/attackby(obj/item/slimepotion/transference/I, mob/user, params)
+	. = ..()
+	if(istype(src, /obj/effect/mob_spawn/human/golem/servant))
+		src.has_owner = FALSE
+	src.flavour_text = null
+	if(iscarbon(user) && can_transfer)
+		var/human_transfer_choice = alert("Transfer your soul to [src]? (Warning, your old body will die!)",,"Yes","No")
+		if(human_transfer_choice != "Yes")
+			return
+		if(QDELETED(src) || uses <= 0)
+			return
+		log_game("[key_name(user)] used [I] to transfer their mind into [src]")
+		user.visible_message("<span class='notice'>As [user] applies the potion on the golem shell, a faint light leaves them, moving to [src] and animating it!</span>","<span class='notice'>You apply the potion to [src], feeling your mind leave your body!</span>")
+		create(ckey = user.ckey, name = user.real_name)
+		user.death()
```

Comment:
This will show the `"X has died at area."` message to anyone observing, which I assume is probably unintentional.

More importantly, the old body is still revivable via defib. I'd suggest changing this to dust them, or maybe delete the old brain once they've transfered over.

Reply:
How would I go about doing the brain deletion? Dusting wouldn't be a good idea since I want them to still have access to their old items.

Reply:
This should work I think:
```dm
	user.death()
	QDEL_NULL(user.internal_organs_slot[BRAIN])
	qdel(I)
```
You might even be able to remove the `death()` since deleting the brain should do that for you.